### PR TITLE
Added feature to ask for user location on map initial load 

### LINKED ISSUE #201: Ask for user location on map initial load
When a user visits the protest map and doesn't have their location cached, we want to get their GPS coordinates.

## 🧐 How to approach:
- The protest map is located in the [`ProtestMap`](https://github.com/guytepper/1km.co.il/blob/master/src/views/ProtestMap/ProtestMap.js) component.
- Check if `userCoordinates` exists in the store - if not ask for the user location using [`getCurrentPosition`](https://github.com/guytepper/1km.co.il/blob/master/src/utils.js#L4-L15). 
- Update the user coordinates in the store, and the protests should be fetched automatically through [the effect in `ProtestMap`](https://github.com/guytepper/1km.co.il/blob/master/src/views/ProtestMap/ProtestMap.js#L23-L29).

### PR DESCRIPTION
**Pull request by omri08**
Resolves #201

### CODE REVIEW

File: src/views/ProtestMap/ProtestMap.js [hunk 1]
```diff
@@ -26,8 +26,19 @@ function ProtestMap() {
...
+  // Ask for user location on map initial load
+  useEffect(() => {
+    const setCoordinates = async () => {
+      if (userCoordinates.length === 0) {
+        const coordinates = await getCurrentPosition();
+        store.setCoordinates(coordinates);
+      }
+    };
+    setCoordinates();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (guytepper) [inline src/views/ProtestMap/ProtestMap.js:42]: Add spacing
```suggestion

    // eslint-disable-next-line react-hooks/exhaustive-deps
```

File: src/views/ProtestMap/ProtestMap.js [hunk 2]
```diff
@@ -26,8 +26,19 @@ function ProtestMap() {
...
+  }, []);
```
Comment (guytepper) [inline src/views/ProtestMap/ProtestMap.js:43]: Space 🛰
```suggestion
  }, []);
  
```

File: src/views/ProtestMap/ProtestMap.js [hunk 3]
```diff
@@ -26,8 +26,19 @@ function ProtestMap() {
...
 
+  // Ask for user location on map initial load
+  useEffect(() => {
+    const setCoordinates = async () => {
+      if (userCoordinates.length === 0) {
+        const coordinates = await getCurrentPosition();
+        store.setCoordinates(coordinates);
+      }
+    };
+    setCoordinates();
```
Comment (guytepper) [inline src/views/ProtestMap/ProtestMap.js:40]: Spacing 👽
```suggestion

    setCoordinates();
```

**Review by guytepper — CHANGES_REQUESTED** [review]
Thanks - it works great!
Just add some spacing so it'll be easier to read 🙂

**Comment by omri08** [comment]
Done :)

**Review by guytepper — APPROVED** [review]

**Pull request merged**